### ISSUE TITLE
AI Changelog Generator 

### ISSUE THREAD

**Issue by demirbey05** [devflex-pro/01-2025-Challenge#4]
### Task Title: AI Changelog Generator 

**Description**:
Create a system that generates a changelog automatically using a Large Language Model (LLM) and online Git tools (BitBucket, GitHub). The system should fetch commit messages from the online Git provider for a specified version improvement period, send them to the LLM for changelog generation, and finally post the changelog to Slack.

**Requirements**:
- As a user, I need to see generated previous backlog messages, so you should store them in some DB.
- As a user, I need to generate changelog and automatically send them to specified Slack group.

**Difficulty**: Medium

**Input and Output Data**:

**Expected Deliverables**:
- CLI tool with API server.
- Unit tests, documentation, and examples.

**Comment by demirbey05** [comment]
Work in progress.

**Comment by justmiles** [comment]
This looks fun! The requirements referenced a database and Slack. "Best practice" is subjective here but I understand it is generally  best practice to just [keep a changelog](https://keepachangelog.com) at the root of the repository - with the source itself. Would it be better if the tool just produced/updated this changelog file?

There could be a subsequent Slack project/app that pulls the actual `changelog.md` from a Git vendor and provides the content between two release versions.

**Comment by demirbey05** [comment]
Yes, we can also use docker, for example if I push docker image now to hub, you can get whole commits that are between today and previous last push date, then issue changelog.

**Comment by devflex-pro** [comment]
Hi. Yes, this looks like a cool project. With all that said, I see it as something that has git data sources, LLM and messengers interfaces, allowing different implementations for git (github, gitlab, bitbicket), messengers (telegram, slack, email), LLM (ChatGPT, etc).

I'll figure it out later, thanks for task)

**Comment by devflex-pro** [comment]
> Yes, we can also use docker, for example if I push docker image now to hub, you can get whole commits that are between today and previous last push date, then issue changelog.

containerizing is default option for my go projects)

**Comment by devflex-pro** [comment]
https://github.com/devflex-pro/AI-Changelog-Generator  

I quickly sketched out a draft, as far as I understood the problem. there are two options, cli and server, it doesn't matter what, but it works like this, roughly speaking, we say where and what commits to take, make a propmpt, send it to LLM, get a response, send a changelog to the destination, it doesn't matter if it's a file, slack, email. for each interface git-provider, llm-provider, destination-provider you can write implementations for any taste.

**Comment by demirbey05** [comment]
> what commits to take

Better to maximize automation here as I said before. At first we can enter manually but glad to have auto-versioning with auto commit capture.
